### PR TITLE
Text To PDF

### LINKED ISSUE #177: [New Feature] Convert text to PDF
Add option to convert a webpage to PDF.
The process should be like this:
- Add a new option in the navigation drawer `Text to PDF`
- There should be an option to select text file
- On clicking it, the file picker should open up
- When PDF is created, show the snackbar (similar to what happens on home page)

Reference:
https://developers.itextpdf.com/content/itext-7-examples/itext-7-differences-itext-5/text-pdf

### PR DESCRIPTION
**Pull request by shiv07tiwari**
# Description

Please include a summary of the change and which issue is fixed. List any dependencies that are required for this change.
If there are any UI change, please include the screenshots also.

Fixes #177

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [ ] `./gradlew assembleDebug assembleRelease`
- [ ] `./gradlew checkstyle`

# Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

**Comment by Swati4star** [comment]
@shiv07tiwari Where is the problem?

**Comment by shiv07tiwari** [comment]
When I am passing the path of text file into BufferReader, I checked if the file exists using file.exists()..and it returns false...the pdf created is unsupported type.
@Swati4star

**Comment by Swati4star** [comment]
@shiv07tiwari I followed this link : http://www.zoftino.com/how-to-create-browse-files-option-in-android

I am successfully able to read the content of txt file as string. I think, you can create the PDF using the content. I have pushed code the in a new branch, in case, you get stuck, https://github.com/Swati4star/Images-to-PDF/commit/198c3aab00f645e5f36aa1e38229bd98dd5a1382

**Comment by shiv07tiwari** [comment]
@Swati4star  FIxed the issue

File: app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        document.open();
+        Font myfont = new Font();
+        myfont.setStyle(Font.NORMAL);
+        myfont.setSize(11);
+
+        document.add(new Paragraph("\n"));
+        readTextFile(fileURI, document, myfont);
+        document.close();
+    }
+    private void readTextFile(Uri uri, Document document, Font myfont) {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java]: add javadoc comments

File: app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+                para.setAlignment(Element.ALIGN_JUSTIFIED);
+                document.add(para);
+            }
+            reader.close();
+            inputStream.close();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+    private void textFileSelect() {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java]: add comments to newly added functions

File: app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+        Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
+        intent.setType("*/*");
+        intent.addCategory(Intent.CATEGORY_OPENABLE);
+
+        try {
+            startActivityForResult(
+                    Intent.createChooser(intent, "Select a File to Upload"),
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java]: add constant string to strings.xml

File: app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+                    FILE_SELECT_CODE);
+        } catch (android.content.ActivityNotFoundException ex) {
+            Snackbar.make(Objects.requireNonNull(getActivity()).findViewById(android.R.id.content),
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java]: Don't use getActivity() function. Create a new private variable `private Activity mActivity;` and assign it value in `onAttach` (just like other fragments. See MergeFilesFragment)

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -229,4 +229,13 @@
     <string name="new_dir">New Directory</string>
     <string name="select_all">Select All</string>
     <string name="converting_file_message">Converting your files to PDF.</string>
+
+    <!-- TODO: Remove or change this placeholder text -->
+    <string name="hello_blank_fragment">Hello blank fragment</string>
```
Comment (Swati4star) [inline app/src/main/res/values/strings.xml]: no need of this blank fragment string

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -229,4 +229,13 @@
...
+    <string name="select_text_file">Select Text File</string>
+    <string name="selected_file_name">Selected File :</string>
+    <string name="no_text_file_selected">No Text File Selected</string>
+    <string name="install_file_manager">Please install a File Manager</string>
+    <string name="text_to_pdf">Text To PDF</string>
+    <string name="text_file_selected">TextFile Selected</string>
```
Comment (Swati4star) [inline app/src/main/res/values/strings.xml]: add space between text & file

File: app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+import java.util.Objects;
+
+import swati4star.createpdf.R;
+import swati4star.createpdf.util.StringUtils;
+
+import static android.app.Activity.RESULT_OK;
+
+public class TextToPdfFragment extends Fragment {
+    private static final int FILE_SELECT_CODE = 0;
+    private static Uri TEXT_FILE_URI = null;
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java]: Do we need to make this variable static?

**Comment by Swati4star** [comment]
@shiv07tiwari I have reviewed the code. Look at the comments.

File: app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+import java.io.InputStreamReader;
+import java.util.Objects;
+
+import swati4star.createpdf.R;
+import swati4star.createpdf.util.StringUtils;
+
+import static android.app.Activity.RESULT_OK;
+
+public class TextToPdfFragment extends Fragment {
+    private static final int FILE_SELECT_CODE = 0;
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java]: add a line before this.

File: app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View rootview =  inflater.inflate(R.layout.fragment_text_to_pdf, container, false);
+
+        Button selectButton = (Button) rootview.findViewById(R.id.selectFile);
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java]: (Button) typecasting is redundant. You can remove that

File: app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java (hunk 2 above)
Comment (Swati4star) [reply app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java]: also change the name to `selectTextFile` or `createTextFileChooser`

File: app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+        selectButton.setOnClickListener( new View.OnClickListener() {
+
+            @Override
+            public void onClick(View v) {
+                textFileSelect();
+                Snackbar.make(Objects.requireNonNull(getActivity()).findViewById(android.R.id.content),
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java]: don't use getActivity()

**Comment by shiv07tiwari** [comment]
@Swati4star  Made the required changes.

File: app/src/main/java/swati4star/createpdf/util/PDFUtils.java [hunk 11]
```diff
@@ -251,6 +259,54 @@ public void onClick(View v) {
...
+     * Read the Text File and put it in document
+     *
+     * @param uri URL to create PDF
+     * @param document PDF Document
+     * @param myfont Font style in PDF
+     */
+    private static void readTextFile(Uri uri, Document document, Font myfont) {
+        InputStream inputStream;
+        try {
+            inputStream = mContext.getContentResolver().openInputStream(uri);
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/PDFUtils.java]: mContext will be null here

**Comment by shiv07tiwari** [comment]
@Swati4star Made the changes.
I am here to learn, and mistakes are gonna help me in that. :)

**Pull request merged**